integr8ly/grafana-operator
# Fix heights

provides a fix for #93 

Adds a compatibility option (spec.compat.fixHeights) for dashboards that have panels with a `height` property encoded as a number. The Grafana SDK expects a string here. Also, newer Grafana versions don't seem to emit this property in exported dashboards anymore.

This is provided as a switch because it manipulates the json before submitting and we'd usually try to avoid this.

Verification steps:

* take the dashboard from this comment and add it to a CR: https://github.com/integr8ly/grafana-operator/issues/93#issuecomment-567498757
* deploy grafana and try to import the CR. It should fail.
* update the `Grafana` CR and add

```
...
spec:
  compat:
    fixHeights: true
...
```

* after a while (max. 30 seconds) the dashboard should get imported successfully
